lib/IDE/Formatting.cpp

```diff
@@ -88,6 +88,14 @@ class FormatContext {
     return InCommentLine;
   }
 
+  bool isSwitchControlStmt(unsigned LineIndex, StringRef Text) {
+    if (!isSwitchContext())
+      return false;
+    StringRef LineText = swift::ide::getTrimmedTextForLine(LineIndex, Text);
+    return LineText.startswith("break") || LineText.startswith("continue") ||
+      LineText.startswith("return");
+  }
```

Comment:
Are we being exhaustive here? IIRC, we can also have at this point a `fallthrough` keyword that we should also indent.